lib/common/socketpool.c

```diff
@@ -376,7 +376,7 @@ static void on_connect(h2o_socket_t *sock, const char *err)
         }
         __sync_sub_and_fetch(&req->pool->_shared.count, 1);
         req->sock = NULL;
-        errstr = "connection failed"; /* shouldn't we return err? */
+        errstr = err;
```

Comment:
This looks like a fine change, though I think we can just get rid of `errstr`, and pass `err` as the argument to `call_connect_cb`?